test/optimizer/stats_catalog_test.cpp

```diff
@@ -0,0 +1,220 @@
+          select_pr->AccessWithNullCheck(all_cols_prm.at(catalog::postgres::STAATTNUM_COL_OID)));
+      EXPECT_TRUE(col_oid != nullptr);
+
+      // Ensure this is one of the columns we created previously
+      EXPECT_EQ(col_oids.count(*col_oid), 1);
+
+      // Insert so that we can compare column OIDs later
+      index_col_oids.insert(*col_oid);
+
+      // Prepare to update this tuple with some bogus statistics
+      auto *const redo = txn->StageWrite(db_, catalog::postgres::STATISTIC_TABLE_OID, all_cols_pri);
+      redo->SetTupleSlot(slot);
+
+      // Write the attributes in the Redo Record
+      auto relid_entry = reinterpret_cast<catalog::table_oid_t *>(
+          redo->Delta()->AccessForceNotNull(all_cols_prm[catalog::postgres::STARELID_COL_OID]));
+      auto attnum_entry = reinterpret_cast<catalog::col_oid_t *>(
+          redo->Delta()->AccessForceNotNull(all_cols_prm[catalog::postgres::STAATTNUM_COL_OID]));
+      auto nullfrac_entry = reinterpret_cast<double *>(
+          redo->Delta()->AccessForceNotNull(all_cols_prm[catalog::postgres::STANULLFRAC_COL_OID]));
+      auto distinct_entry = reinterpret_cast<double *>(
+          redo->Delta()->AccessForceNotNull(all_cols_prm[catalog::postgres::STADISTINCT_COL_OID]));
+      auto numrows_entry = reinterpret_cast<uint32_t *>(
+          redo->Delta()->AccessForceNotNull(all_cols_prm[catalog::postgres::STA_NUMROWS_COL_OID]));
+
+      *relid_entry = table_oid;
+      *attnum_entry = *col_oid;
+      *nullfrac_entry = 0.123;
+      *distinct_entry = 4.56;
+      *numrows_entry = 15721;
+
+      result = statistics_->Update(common::ManagedPointer(txn), redo);
+      EXPECT_TRUE(result);
+    }
+
+    txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  }
+
+  // Ensure the sets of column OIDs are equal
+  EXPECT_EQ(col_oids, index_col_oids);
+
+  // Ensure that GetTableStats returns the correct statistics
+  {
+    auto txn = txn_manager_->BeginTransaction();
+    auto accessor = catalog_->GetAccessor(common::ManagedPointer(txn), db_);
+    EXPECT_NE(accessor, nullptr);
+
+    auto table_stats = accessor->GetTableStats(table_oid);
+    for (catalog::col_oid_t col_oid : col_oids) {
+      auto col_stats = table_stats->GetColumnStats(col_oid);
+
+      // TODO(khg): no way to get frac_null_ out of a ColumnStats?
+      // EXPECT_EQ(col_stats->GetFracNull(), 0.123);
+      EXPECT_EQ(col_stats->GetCardinality(), 4.56);
+      EXPECT_EQ(col_stats->GetNumRows(), 15721);
+    }
+
+    txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  }
+
+  // Check that pg_statistic tuples are deleted when table is dropped
+  {
+    auto txn = txn_manager_->BeginTransaction();
+    auto accessor = catalog_->GetAccessor(common::ManagedPointer(txn), db_);
+    EXPECT_NE(accessor, nullptr);
+
+    // Drop table from storage
+    accessor->DropTable(table_oid);
+
+    // Search the statistics index for matching tuples
```

Comment:
This comment can be made far better. Plus you can tell what exactly does the function do?